### PR TITLE
cinny: init at 1.6.1

### PR DESCRIPTION
**Pull request by wahjava**
###### Motivation for this change

Cinny is a web client for Matrix, and would like it to be available to nixpkgs. The derivation is architecture independent similar to `element-web`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/cinny/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/
+    cp -R . $out/
+
+    runHook postInstall
+  '';
+
+  meta = {
```
Comment (legendofmiracles) [inline pkgs/applications/networking/instant-messengers/cinny/default.nix:21]: Use with lib; here to avoid the lib. noise further down

Comment (wahjava) [reply pkgs/applications/networking/instant-messengers/cinny/default.nix]: Done. Thanks!

**Comment by wahjava** [comment]
The mentioned version is [vulnerable now](https://matrix.org/blog/2021/09/13/vulnerability-disclosure-key-sharing), so switching the PR to draft mode. I'll submit an update, and un-draft it.

Thanks!

**Comment by ashkitten** [comment]
thanks for doing this! it would probably be better to build cinny ourselves instead of just using the prebuilt release, if that's not too much effort

**Comment by wahjava** [comment]
> thanks for doing this! it would probably be better to build cinny ourselves instead of just using the prebuilt release, if that's not too much effort

Sure, although I followed `element-web`'s foot steps :)

**Review by SuperSandro2000 — COMMENTED** [review]
Do we really don't need to do any patching?

**Comment by wahjava** [comment]
> Do we really don't need to do any patching?

AFAICS none, and also added support for configuration file which is implemented in 1.5.0.

**Comment by peperunas** [comment]
Hello, sorry to jump in the PR. What is the current status of the PR? Is it going to be merged?

**Review by legendofmiracles — APPROVED** [review]
Could chat with my other account by building it and then `python3 -m http.server` serving it with a simple http server

**Comment by wahjava** [comment]
You sir are a @legendofmiracles :smile:  Thanks!

**Pull request merged**